rosbag2/src/rosbag2/writer.cpp

```diff
@@ -105,4 +148,21 @@ bool Writer::should_split_bagfile() const
     return storage_->get_bagfile_size() > max_bagfile_size_;
   }
 }
+
+void Writer::finalize_metadata()
+{
+  metadata_.bag_size = 0;
+
+  for (const auto & path : metadata_.relative_file_paths) {
+    metadata_.bag_size += rosbag2_storage::FilesystemHelper::get_file_size(path);
+  }
+
+  metadata_.topics_with_message_count.clear();
```

Comment:
should that maybe be done in the open function, when metadata_ is being initialized? I mean, actually, I would expect we don't need that call.

Reply:
Exact same reason as [the previous comment](https://github.com/ros2/rosbag2/pull/184#discussion_r338328148); this line should be a no-op. 

I'm just being explicit that `metadata_.topics_with_message_count` will only hold the values from `topics_`.